### PR TITLE
Add support for setting app version from SHA for test catalogs

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/13206

Needed for adding integration test in https://github.com/giantswarm/app-exporter/pull/9.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: apptest.go [hunk 1]
```diff
@@ -191,3 +197,24 @@ func (a *AppSetup) waitForDeployedApp(ctx context.Context, appName string) error
 
 	return nil
 }
+
+// getVersionForApp checks whether a commit SHA or a version was provided.
+// If a SHA was provided then we check the test catalog to get the latest version.
+// As for test catalogs the version format used is [latest version]-[sha].
+// e.g. 0.2.0-ad12c88111d7513114a1257994634e2ae81115a2
```
Comment (tomahawk28) [inline apptest.go:204]: I think we need additional comment about no SHA was provided. WDYT?
```suggestion
// e.g. 0.2.0-ad12c88111d7513114a1257994634e2ae81115a2
// If no SHA was provided then we give the latest version in our official catalog. 
```

Comment (rossf7) [reply apptest.go:204]: Good idea. I extended the comment. It just uses the version provided. This is for when we don't want to install from a test catalog. WDYT?

Comment (tomahawk28) [reply apptest.go:204]: Nice, 👍

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**